Recipes with extra spaces causes java.lang.StringIndexOutOfBoundsException

```scala
      Items.STICK + Items.DIAMOND to Items.DIAMOND_SWORD withShape
        """
          | ..d
          | .d.
          | s..
        """.stripMargin,
```

Exception:

```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 11
	at java.lang.String.charAt(String.java:658)
	at net.minecraft.item.crafting.CraftingManager.addRecipe(CraftingManager.java:270)
	at net.minecraftforge.fml.common.registry.GameRegistry.addShapedRecipe(GameRegistry.java:282)
	at net.minecraftforge.fml.common.registry.GameRegistry.addRecipe(GameRegistry.java:277)
	at com.easyforger.recipes.RecipeSupport$$anonfun$crafting$1.apply(RecipeSupport.scala:44
```

--- a/mods/src/main/scala/com/easyforger/samples/items/ItemsMod.scala
+++ b/mods/src/main/scala/com/easyforger/samples/items/ItemsMod.scala
@@ -59,9 +59,9 @@ object ItemsMod extends EasyForger {
         """.stripMargin,
       Items.STICK + Blocks.TNT to explosionRod withShape
         """
-          |..t
-          |.s.
-          |s..
+          | ..t
+          | .s.
+          | s..
         """.stripMargin,
       Items.IRON_SWORD + Items.POISONOUS_POTATO to venomSword,
       Items.IRON_PICKAXE + Items.DIAMOND to quickPick,

--- a/src/main/scala/com/easyforger/recipes/RecipeSupport.scala
+++ b/src/main/scala/com/easyforger/recipes/RecipeSupport.scala
@@ -83,7 +83,9 @@ object RecipeSupport {
     }
 
   def calcMCParamsArray(craftRecipe: CraftingRecipe): Array[Object] = {
-    val params = craftRecipe.shape.get.trim.replace('.', ' ').split("\n")
+    val params = craftRecipe.shape
+      .map(_.trim.replace(" ", "").replace('.', ' ').split("\n"))
+      .getOrElse(Array.empty)
 
     // turn into a list to avoid removing duplicated acronyms, which would end up hiding errors
     val acronyms = craftRecipe.sources.toList

--- a/tester/src/main/scala/com/easyforger/recipes/test/ShapedRecipesTest.scala
+++ b/tester/src/main/scala/com/easyforger/recipes/test/ShapedRecipesTest.scala
@@ -23,7 +23,7 @@ object ShapedRecipesTest {
           """.stripMargin
 
         val params = RecipeSupport.calcMCParamsArray(recipe)
-        assertParam(params(0), "   ")
+        assertParam(params(0), "   ") // scalastyle:ignore
         assertParam(params(1), "lll")
         assertParam(params(2), "i i")
         assertParam(params(3), 'l')
@@ -67,6 +67,23 @@ object ShapedRecipesTest {
         assertSameStackAs(params(4).asInstanceOf[ItemStack], new ItemStack(Items.IRON_INGOT))
         assertParam(params(5), 'r')
         assertSameStackAs(params(6).asInstanceOf[ItemStack], redDye)
+      }
+
+      "ignore leading and trailing spaces" - {
+        val recipe = Items.SIGN to Items.APPLE withShape
+          """
+            | s.s
+            | .s.
+            | ...
+          """.stripMargin
+
+        val params = RecipeSupport.calcMCParamsArray(recipe)
+
+        assertParam(params(0), "s s")
+        assertParam(params(1), " s ")
+        assertParam(params(2), "   ")
+        assertParam(params(3), 's')
+        assertSameStackAs(params(4).asInstanceOf[ItemStack], new ItemStack(Items.SIGN))
       }
     }
   }

